test/gtest/uct/test_mem.cc

```diff
@@ -118,6 +118,8 @@ UCS_TEST_P(test_mem, md_alloc) {
                 params.flags  |= UCT_MD_MEM_ACCESS_ALL;
                 params.mds.mds = &md;
 
+                mem_buffer::set_device_context();
```

Comment:
can we set it from https://github.com/openucx/ucx/blob/master/test/gtest/common/main.cc#L55 main function?
it is done before running all tests and can be used for global initialization

Reply:
@yosefe Can do that but we'd be setting a device context for all tests. There is no harm in doing that as I see it but is it ok?

Reply:
if we set just in this test, it will be to the ~50% of the test which would run after it. and better have a predictable behavior that does not depend on tests order